### PR TITLE
persisting nvi report when nvi reported cristin object

### PR DESCRIPTION
**Pull request by truhacevkir**

### CODE REVIEW

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java [hunk 1]
```diff
@@ -173,9 +178,50 @@ private EventReference getEventReferenceFromBody(String body) {
     private Publication persistConversionReports(PublicationRepresentations publicationRepresentations) {
         persistCristinIdentifierInFileNamedWithPublicationIdentifier(publicationRepresentations);
         persistPartOfCristinIdentifierIfPartOfExists(publicationRepresentations);
+        persistNviReportIfNeeded(publicationRepresentations);
         return publicationRepresentations.getPublication();
     }
 
+    private void persistNviReportIfNeeded(PublicationRepresentations publicationRepresentations) {
```
Comment (anetteOlli) [inline cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java:185]: Beyond the persistNviReportNeeded methods should be in it's own class.

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinLocale.java [hunk 2]
```diff
@@ -55,6 +58,16 @@ public class CristinLocale {
...
+    @JsonProperty(CONTROLLED_BY_FIELD)
+    private String controlledBy;
+
+    @JsonProperty(DATE_CONTROLLED_FIELD)
+    private String dateControlled;
+
+    @JsonProperty(CONTROL_STATUS_FIELD)
+    private String controlStatus;
+
+
```
Comment (anetteOlli) [inline cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinLocale.java:70]: You could make a method toNVIReportFormat here.

File: cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumerTest.java [hunk 3]
```diff
@@ -175,6 +176,23 @@ void shouldPersistCristinIdInFileNamedWithPublicationIdentifier() throws IOExcep
...
+        var publications = handler.handleRequest(sqsEvent, CONTEXT);
+        var actualPublication = publications.get(0);
+        var expectedFileNameStoredInS3 = actualPublication.getIdentifier().toString();
+
+        var expectedTimestamp = eventBody.getTimestamp();
+        var expectedFileLocation = NVI_FOLDER
+                                            .addChild(timestampToString(expectedTimestamp))
+                                            .addChild(expectedFileNameStoredInS3);
+
+        assertDoesNotThrow(() -> s3Driver.getFile(expectedFileLocation));
```
Comment (anetteOlli) [inline cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumerTest.java:193]: This is a bit on the thin side. I would like some asserts that you can parse the data from the report and that the data contains what you except.

**Review by anetteOlli — CHANGES_REQUESTED** [review]

File: scopus-import/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java [hunk 4]
```diff
@@ -704,6 +707,202 @@ void shouldThrowExceptionWhenSrcTypeIsNotSupported() throws IOException {
...
+                "status": 400,
+                "detail": "Invalid path parameter for identifier, needs to be a number or an ORCID",
+                "requestId": "020e9c34-7d7d-4535-98af-1a13e2a23d7c"
+            }
+            """;
+        return cristinPerson0response;
+    }
+
+    private static String pia7005704879response() {
+        @org.intellij.lang.annotations.Language("json")
```
Comment (anetteOlli) [inline scopus-import/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java:742]: Dette bør leses fra en fil.

File: scopus-import/src/test/resources/2-s2.0-85178336140.xml [hunk 5]
```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (anetteOlli) [inline scopus-import/src/test/resources/2-s2.0-85178336140.xml:1]: Brukes denne? i så fall gi den et navn som dokumenterer hva den tester.

**Review by anetteOlli — CHANGES_REQUESTED** [review]

File: scopus-import/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java (hunk 4 above)
Comment (truhacevkir) [reply scopus-import/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java:742]: fjernet

File: scopus-import/src/test/resources/2-s2.0-85178336140.xml (hunk 5 above)
Comment (truhacevkir) [reply scopus-import/src/test/resources/2-s2.0-85178336140.xml:1]: fjernet

**Review by anetteOlli — APPROVED** [review]

**Pull request merged**